packages/core/src/plugins/html-deserializer/utils/htmlTextNodeToString.ts

```diff
@@ -3,9 +3,101 @@
  */
 import { isHtmlText } from './isHtmlText';
 
-export const htmlTextNodeToString = (node: HTMLElement | ChildNode) => {
+function getStyleFromNode(node: HTMLElement | ChildNode): string {
+  if (!(node as HTMLElement).getAttribute) return '';
+
+  const styleAttr = (node as HTMLElement).getAttribute('style');
+
+  return styleAttr ? styleAttr.trim() : '';
+}
+
+// Can use the npm package style-to-object.
+// https://www.npmjs.com/package/style-to-object
+function styleToObject(styleText: string): Record<string, string> {
+  const styleObject = {};
+  const declarations = styleText.split(';');
+
+  const camelCase = (str: string) => {
+    return str.replaceAll(/-([a-z])/g, (match: string, letter: string) => {
+      return letter.toUpperCase();
+    });
+  };
+
+  declarations.forEach((declaration) => {
+    declaration = declaration.trim();
+    if (declaration) {
+      const [property, value] = declaration.split(':');
+      const propertyName = camelCase(property.trim());
+      // @ts-ignore
+      styleObject[propertyName] = value.trim();
+    }
+  });
+
+  return styleObject;
+}
```

Comment:
The `style` attribute only produces results when the DOM is actually inserted into the `window.document`. Before being inserted into the page, it will always be an empty string.

Reply:
Got it. That explains some weird results I saw while testing. :+1:

Reply:
One thing I've just noticed is we're already using `.style` in other parts of the deserialization pipeline. For example, in `createParagraphPlugin`, we check if the paragraph element has a particular font-family using `el.style.fontFamily`. The element object comes from (in reverse order): `pluginDeserializeHtml`, `pipeDeserializeHtmlElement`, `htmlElementToElement` and `deserializeHtmlNode`, the last of which is the function that also calls `htmlTextNodeToString`. Any idea why `style` is working in `createParagraphPlugin` but not here?

Reply:
I tested this further, and here's what I've found:

- `el.style` always returns a CSSProperties object matching the literal `style` attribute on a HTML element
  - This works regardless of whether the element has been inserted into the DOM
  - The default value for missing styles is the empty string
  - The `style` property for a `<pre>` tag does not contain `whiteSpace` unless this is set explicitly
- `window.getComputedStyle(el)` also returns a CSSProperties object
  - This **only works** if the element has been inserted into the DOM
  - The computed `whiteSpace` property for a `<pre>` element is `'pre'`

Inserting the element into the DOM prior to checking its style isn't a viable solution (since this would evaluate any script tags). However, it looks like the normal `style` property works the same way as your `styleToObject` function, while also being more strongly typed.

Can you check to see if replacing `styleToObject` with just `node.style` works here?

Reply:
This is my mistake, I realize it is possible.